src/App.tsx

```diff
@@ -4,67 +4,114 @@ import Signup from './modules/Signup';
+                    <Signup />
+                </Protected>
         },
         {
             path: "/login",
-            element: <Login />
-        }, 
+            element:
+                <Protected authentication={false}>
+                    <Login />
+                </Protected>
+        },
         {
             path: "/",
-            element:  <Home/>
+            element:
+                <Protected authentication={false}>
+                  <Landing/>
+                </Protected>
         },
         {
             path: "/admin",
-            element:  <AdminDashBoard/>
+            element:
+                <Protected authentication={true}>
+                    <AdminDashBoard />
+                </Protected>
         },
         {
-            path:"/viewproducts",
-            element:<ViewProducts onProductCountChange={() => {}} />
-
+            path: "/viewproducts",
+            element: <Protected authentication={true}>
+                      <ViewProducts />
+                     </Protected>
+        },
+        {
+            path: "/getusers",
+            element: <Protected authentication={true}>
+                      <ViewUsers />
+                     </Protected>
+        },
+        {
+            path: "/getorders",
+            element: <Protected authentication={true}><ViewOrders /></Protected>
+        },
+        {
+            path: "/userpage",
+            element: <Protected authentication={true}><UserPage /></Protected>
+        },
+        {
+            path: "/buyproduct",
+            element: <Protected authentication={true}><BuyProducts /></Protected>
         },
         {
-            path:"/getusers",
-            element:<ViewUsers/>
+            path: "*",
+            element: <Protected authentication={true}><Error /></Protected>
         },
         {
-            path:"/getorders",
-            element:<ViewOrders/>
+            path: "/myorders",
+            element: <Protected authentication={true}><UserOrders /></Protected>
         },
         {
-            path:"/userpage",
-            element:<UserPage/>
+            path: "/mycart",
+            element:  <Protected authentication={true}><Cart/></Protected>
         },
         {
-            path:"/buyproduct",
-            element:<BuyProducts/>
+            path: "/confirmorder",
+            element:  <Protected authentication={true}><Orderconfirm/></Protected>
         },
         {
-            path:"*",
-            element:<Error/>
+            path: "/footer",
+            element: <Protected authentication={true}><Footer/></Protected>
         },
 
     ])
     return (
-        <QueryClientProvider client={queryClient}>
-        <div className="App">
-            <RouterProvider router={router} />
-        </div>
-        </QueryClientProvider>
+        <>
+        <ToastContainer/>
+        <Provider store={store}>
+            <QueryClientProvider client={queryClient}>
```

Comment:
Why are you using both. React-query & Redux?
Only one should have been used. You can use Redux toolkit query OR Just React query. We never use 2 global state management libraries in a single project